core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

```diff
@@ -301,4 +302,47 @@ public void testEmptyQuery() throws IOException {
         String query = "{ \"query\": {} }";
         assertParseSearchSource(builder, new BytesArray(query), ParseFieldMatcher.EMPTY);
     }
+
+    public void testParseIndicesBoost() throws IOException {
+        {
+            String restContent = " { \"indices_boost\": {\"foo\": 1.0, \"bar\": 2.0}}";
+            try (XContentParser parser = XContentFactory.xContent(restContent).createParser(restContent)) {
+                SearchSourceBuilder searchSourceBuilder = SearchSourceBuilder.fromXContent(createParseContext(parser),
+                    searchRequestParsers.aggParsers, searchRequestParsers.suggesters, searchRequestParsers.searchExtParsers);
+                assertEquals(2, searchSourceBuilder.indexBoost().size());
+                assertEquals(new Tuple<>("foo", 1.0f), searchSourceBuilder.indexBoost().get(0));
+                assertEquals(new Tuple<>("bar", 2.0f), searchSourceBuilder.indexBoost().get(1));
+            }
+        }
+
+        {
+            String restContent = "{" +
+                "    \"indices_boost\" : [\n" +
+                "        { \"foo\" : 1.0 },\n" +
+                "        { \"bar\" : 2.0 },\n" +
+                "        { \"baz\" : 3.0 }\n" +
+                "    ]}";
+            try (XContentParser parser = XContentFactory.xContent(restContent).createParser(restContent)) {
+                SearchSourceBuilder searchSourceBuilder = SearchSourceBuilder.fromXContent(createParseContext(parser),
+                    searchRequestParsers.aggParsers, searchRequestParsers.suggesters, searchRequestParsers.searchExtParsers);
+                assertEquals(3, searchSourceBuilder.indexBoost().size());
+                assertEquals(new Tuple<>("foo", 1.0f), searchSourceBuilder.indexBoost().get(0));
+                assertEquals(new Tuple<>("bar", 2.0f), searchSourceBuilder.indexBoost().get(1));
+                assertEquals(new Tuple<>("baz", 3.0f), searchSourceBuilder.indexBoost().get(2));
+            }
+        }
+
+        {
+            String restContent = "{" +
+                "    \"indices_boost\" : [\n" +
+                "        { \"foo\" : 1.0, \"bar\": 2.0}\n" + // invalid format
+                "    ]}";
+
+            try (XContentParser parser = XContentFactory.xContent(restContent).createParser(restContent)) {
+                ParsingException e = expectThrows(ParsingException.class, () -> SearchSourceBuilder.fromXContent(createParseContext(parser),
+                    searchRequestParsers.aggParsers, searchRequestParsers.suggesters, searchRequestParsers.searchExtParsers));
+                assertEquals("Expected [END_OBJECT] in [indices_boost] but found [FIELD_NAME]", e.getMessage());
```

Comment:
thanks for adding these tests! do we want to add a few more with all the exception cases that you handled in the parser?